### ISSUE TITLE
TileGroup component for Radio Tiles, fieldset requires legend

### ISSUE THREAD

**Issue by jaksart1** [carbon-design-system/carbon#6941]
## Environment

> Operating system

macOS

> Browser

Firefox/Chrome

> Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

## Detailed description

> What version of the Carbon Design System are you using?

carbon-components-react 7.18.0

> What did you expect to happen?
 
Expected no accessibility issues when excluding optional `legend` prop in `TileGroup`

> What happened instead?

IBM Accessibility checker shows a violation.

Because the TileGroup component creates a fieldset, the Accessibility checker requires a legend inside. We need a way to not have a legend for the TileGroup (or have it hidden), or a way to assign the legend to a different element that is outside the Tile Group.

> What WCAG 2.1 checkpoint does the issue violate?

`<fieldset>` element does not have a `<legend>`

https://www.w3.org/WAI/WCAG21/Techniques/html/H71

> - Style and vanilla JS:
>   https://codesandbox.io/s/github/carbon-design-system/carbon/tree/master/packages/components/examples/codesandbox
> - React:
>   https://codesandbox.io/s/github/carbon-design-system/carbon/tree/master/packages/react/examples/codesandbox

## Additional information

Our team has a set of Radio Tiles inside a TileGroup and we do not want a legend, however when we remove the legend we get an accessibility violation.

For IBM Code Engine

- Screenshots or code
- Notes

**Comment by tw15egan** [comment]
Since a TileGroup containing RadioTiles needs a label to pass a11y tests (it is a group of radio buttons under the hood, which require a legend), I'm not sure it's possible to remove the label without causing a violation. The `legend` prop might actually need to be changed to `required` because of this.

cc @dakahn
